### DISCUSSION TITLE
支持自定义后端是否可以展开讲讲？

### DISCUSSION THREAD

**Discussion by randy-chiu** [mlc-ai/mlc-zh#146] [Q&A] [answered]
当前tvm已支持多种自定义后端硬件指令，课程8描述了一些针对自定义硬件后端的整体思路，具体点的示例有没有材料，比如如何映射到具体硬件指令等？或者有没有相关的示例代码可以参考的，比如tvm中？

**Comment by yzh119** [comment, marked as answer]
不同后端的指令集定义：https://github.com/apache/tvm/tree/1ecf084eecaff167967df1a8c998de72e1198c24/python/tvm/tir/tensor_intrin
对应的代码生成逻辑（以cuda为例）：https://github.com/apache/tvm/blob/1ecf084eecaff167967df1a8c998de72e1198c24/src/target/source/codegen_cuda.cc#L675-L906

**Reply by randy-chiu** [reply]
谢谢，这个信息对我很有用！
想继续问下，除了tvm中的相关源码外，有没有其它资料/信息可以对生成自定义后端比较有帮助的？只看这两个源码对于理解后端整体的执行流程仍然有点困难，谢谢。

**Reply by yzh119** [reply]
[driver_api.cc](https://github.com/apache/tvm/blob/79c48f38878788b46d3acd1945469ae97e508d7d/src/driver/driver_api.cc) 这个函数基本上描述了从TensorIR到后端代码的生成的逻辑，可以手动打断点来理解每个pass的语义。

**Reply by randy-chiu** [reply]
好的，我看下相关代码学习下，谢谢！
